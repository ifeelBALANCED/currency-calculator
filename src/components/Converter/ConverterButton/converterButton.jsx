import React from 'react';
import {Button} from "@chakra-ui/react";

const ConverterButton = ({convertHandler}) => {
    return (
        <Button
            mt={4}
            onClick={() => convertHandler()}
            colorScheme="teal"
        >
            Convert
        </Button>
    );
};

export default ConverterButton;
