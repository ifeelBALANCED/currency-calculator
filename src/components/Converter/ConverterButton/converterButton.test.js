import React from "react";
import ConverterButton from "./converterButton";
// set up component
const setUp = () => shallow(<ConverterButton/>)
// describe wrapper
describe('should render ConverterButton component', () => {
    let component
    // In the case when you need to do the same work for several tests, you can use the beforeEach functions
    beforeEach(() => {
        component = setUp()
    })
    // test for that check button render correctly
    it('should render button component', () => {
        expect(component).toMatchSnapshot()
    });
    describe('it should render ConvertButton component', () => {
        describe('ConvertButton handlers', () => {
            // test onChange func in component
            it('should call onClick method', () => {
                const mockClick = jest.fn()
                // mount return the wrapper instance around the rendered output
                const component = mount(<ConverterButton convertHandler={mockClick}/>)
                const btn = component.find('button')
                btn.simulate('click')
                expect(mockClick.mock.calls.length).toEqual(1)
            });
        });
    })
});
