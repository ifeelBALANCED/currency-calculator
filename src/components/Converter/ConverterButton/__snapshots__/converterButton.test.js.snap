// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ConverterButton component should render button component 1`] = `
<Button
  colorScheme="teal"
  mt={4}
  onClick={[Function]}
>
  Convert
</Button>
`;
